cloudinit/config/cc_set_passwords.py

```diff
@@ -5,85 +5,65 @@
-on-disk unchanged.
+to accept password authentication.
 
 The ``chpasswd`` config key accepts a dictionary containing either or both of
-``expire`` and ``list``.
-
-If the ``list`` key is provided, it should contain a list of
-``username:password`` pairs.  This can be either a YAML list (of strings), or a
-multi-line string with one pair per line.  Each user will have the
-corresponding password set.  A password can be randomly generated by specifying
-``RANDOM`` or ``R`` as a user's password.  A hashed password, created by a tool
-like ``mkpasswd``, can be specified; a regex
-(``r'\\$(1|2a|2y|5|6)(\\$.+){2}'``) is used to determine if a password value
-should be treated as a hash.
-
-.. note::
-    The users specified must already exist on the system.  Users will have been
-    created by the ``cc_users_groups`` module at this point.
-
-By default, all users on the system will have their passwords expired (meaning
-that they will have to be reset the next time the user logs in).  To disable
-this behaviour, set ``expire`` under ``chpasswd`` to a false value.
-
-If a ``list`` of user/password pairs is not specified under ``chpasswd``, then
-the value of the ``password`` config key will be used to set the default user's
-password.
-
-**Internal name:** ``cc_set_passwords``
-
-**Module frequency:** per instance
-
-**Supported distros:** all
-
-**Config keys**::
-
-    ssh_pwauth: <yes/no/unchanged>
-
-    password: password1
-    chpasswd:
-        expire: <true/false>
-
-    chpasswd:
-        list: |
-            user1:password1
-            user2:RANDOM
-            user3:password3
-            user4:R
-
-    ##
-    # or as yaml list
-    ##
-    chpasswd:
-        list:
-            - user1:password1
-            - user2:RANDOM
-            - user3:password3
-            - user4:R
-            - user4:$6$rL..$ej...
-"""
+``list`` and ``expire``. The ``list`` key is used to assign a password to a
+to a corresponding pre-existing user. The ``expire`` key is used to set
+whether to expire all user passwords such that a password will need to be reset
+on the user's next login.
 
-import re
-from string import ascii_letters, digits
+``password`` config key is used to set the default user's password. It is
+ignored if the ``chpasswd`` ``list`` is used.
+"""
 
-from cloudinit import log as logging
-from cloudinit import subp, util
-from cloudinit.distros import ug_util
-from cloudinit.ssh_util import update_ssh_config
+meta: MetaSchema = {
+    "id": "cc_set_passwords",
+    "name": "Set Passwords",
+    "title": "Set user passwords and enable/disable SSH password auth",
+    "description": MODULE_DESCRIPTION,
+    "distros": [ALL_DISTROS],
+    "frequency": PER_INSTANCE,
+    "examples": [
+        dedent(
+            """\
+            ssh_pwauth: true
+            password: password1
+            """
+        ),
+        dedent(
+            """\
+            ssh_pwauth: false
```

Comment:
Do we want a multi-line contextual header comment in these examples to tell people what the snippet's behavior is?